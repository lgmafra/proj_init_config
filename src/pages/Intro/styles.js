import { StyleSheet } from 'react-native';
import { colors, metrics } from '~/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    padding: metrics.basePadding,
  },
  textInfo: {
    height: 135,
  },
  info: {
    paddingLeft: metrics.basePadding,
    fontSize: 36,
    color: colors.white,
  },
  infoBold: {
    paddingLeft: metrics.basePadding,
    fontSize: 36,
    color: colors.white,
    fontWeight: 'bold',
  },
  infoLogin: {
    paddingLeft: metrics.basePadding,
    fontSize: 21,
    color: colors.white,
    fontWeight: '300',
    marginTop: metrics.baseMargin * 2,
    marginBottom: metrics.baseMargin * 2,
  },
  buttonsPlace: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  buttonLogin: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 138,
    height: 55,
    borderRadius: metrics.baseRadius * 9,
    backgroundColor: colors.primary,
  },
  buttonNew: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 138,
    height: 55,
    borderRadius: metrics.baseRadius * 9,
    borderWidth: 2,
    borderColor: colors.white,
  },
  buttonText: {
    fontSize: 18,
    color: colors.white,
  },
  logo: {
    paddingLeft: metrics.basePadding,
    width: 153,
    height: 22,
    marginTop: metrics.baseMargin * 17,
  },
});

export default styles;
