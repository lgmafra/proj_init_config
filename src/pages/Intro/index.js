import React from 'react';
import {
  View, Text, TouchableOpacity, Image, ImageBackground, StatusBar,
} from 'react-native';
import { useNavigation } from 'react-navigation-hooks';

import styles from './styles';

import image from '~/images/image.png';
import logo from '~/images/logo.png';

export default function Intro() {
  const { navigate } = useNavigation();

  return (
    <ImageBackground style={styles.container} resizeMode="cover" source={image}>
      <StatusBar barStyle="light-content" />
      <View style={styles.textInfo}>
        <Text style={styles.info}>Investimento</Text>
        <Text style={styles.info}>imobiliário</Text>
        <Text style={styles.infoBold}>simplificado</Text>
      </View>
      <Text style={styles.infoLogin}>Faça seu login ou crie sua conta</Text>
      <View style={styles.buttonsPlace}>
        <TouchableOpacity style={styles.buttonLogin} onPress={() => navigate('Login')}>
          <Text style={styles.buttonText}>Login</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.buttonNew} onPress={() => {}}>
          <Text style={styles.buttonText}>Criar Conta</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.logo}>
        <Image source={logo} />
      </View>
    </ImageBackground>
  );
}
