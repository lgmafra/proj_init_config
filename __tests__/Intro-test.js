import React from 'react';
import { shallow } from 'enzyme';
import Intro from '../src/pages/Intro';

describe('test Intro page component', () => {
  const mockProps = {
    navigation: { navigate: jest.fn() },
  };

  const wrapper = shallow(<Intro {...mockProps} />);

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('intro page has 2 buttons', () => {
    expect(wrapper.find('TouchableOpacity').length).toBe(2);
  });

  const loginButton = wrapper.findWhere(node => node.prop('testID') === 'buttonLogin');
  const newAccountButton = wrapper.findWhere(node => node.prop('testID') === 'buttonNewAccount');

  it('login button exists', () => {
    expect(loginButton.exists()).toBe(true);
  });

  it('press dont crash', () => {
    loginButton.props().onPress();
  });

  it('new account button exists', () => {
    expect(newAccountButton.exists()).toBe(true);
  });

  it('press dont crash', () => {
    loginButton.props().onPress();
  });
});
