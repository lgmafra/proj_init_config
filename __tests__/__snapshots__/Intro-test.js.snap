// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Intro page component match snapshot 1`] = `ShallowWrapper {}`;
